src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1321,10 +1321,21 @@ private static JsonWebToken ValidateSignature(string token, TokenValidationParam
 
             }
 
+            // Get information on where keys used during token validation came from for debugging purposes.
+            var keysInTokenValidationParameters = TokenUtilities.GetAllSigningKeys(validationParameters);
+            var keysInConfiguration = TokenUtilities.GetAllSigningKeys(configuration);
+            var numKeysInTokenValidationParameters = keysInTokenValidationParameters.Count();
+            var numKeysInConfiguration = keysInConfiguration.Count();
+
             if (kidExists)
             {
                 if (kidMatched)
-                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10511, keysAttempted, jwtToken.Kid, exceptionStrings, jwtToken)));
+                {
+                    var isKidInTVP = keysInTokenValidationParameters.Any(x => x.KeyId.Equals(jwtToken.Kid, StringComparison.Ordinal));
```

Comment:
why not simplify
```suggestion
                    var isKidInTVP = keysInTokenValidationParameters.Any(x => x.KeyId == jwtToken.Kid);
```

Reply:
This allows us to specify the type of StringComparison being used, and so it's more explicit about how the strings are being compared. 

Also, I believe we generally use 'Equals' over '==' in this library even when the types of the variables being compared are strings.

Reply:
@mafurman thank you for the explanation.

I have proposed a separate PR https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1775  to change the convention as operator == makes more sense for me